supertokens_python/recipe/emailverification/recipe.py

```diff
@@ -186,10 +236,164 @@ def get_instance() -> EmailVerificationRecipe:
             "Initialisation not done. Did you forget to call the SuperTokens.init function?"
         )
 
+    @staticmethod
+    def get_instance_optional() -> Optional[EmailVerificationRecipe]:
+        return EmailVerificationRecipe.__instance
+
     @staticmethod
     def reset():
         if ("SUPERTOKENS_ENV" not in environ) or (
             environ["SUPERTOKENS_ENV"] != "testing"
         ):
             raise_general_exception("calling testing function in non testing env")
         EmailVerificationRecipe.__instance = None
+
+    async def get_email_for_user_id(
+        self, user_id: str, user_context: Dict[str, Any]
+    ) -> Union[GetEmailForUserIdOkResult, EmailDoesnotExistError, UnknownUserIdError]:
+        if self.config.get_email_for_user_id is not None:
+            res = await self.config.get_email_for_user_id(user_id, user_context)
+            if not isinstance(res, UnknownUserIdError):
+                return res
+
+        for f in self.get_email_for_user_id_funcs_from_other_recipes:
+            res = await f(user_id, user_context)
+            if not isinstance(res, UnknownUserIdError):
+                return res
+
+        return UnknownUserIdError()
+
+    def add_get_email_for_user_id_func(self, f: Callable[[str, Dict[str, Any]], Any]):
+        self.get_email_for_user_id_funcs_from_other_recipes.append(f)
+
+
+class EmailVerificationClaimClass(BooleanClaim):
```

Comment:
why is this class in this file and not in the ev_claim.py file?

Reply:
I had to move the content to avoid cyclic import errors. It's really troublesome this time. Have already wasted hours. Need help to fix it.